code/modules/surgery/organs/internal/heart/_heart.dm

```diff
@@ -197,35 +207,63 @@
 
 /obj/item/organ/internal/heart/cybernetic/on_life(seconds_per_tick, times_fired)
 	. = ..()
-	if(dose_available && owner.health <= owner.crit_threshold && !owner.reagents.has_reagent(rid))
-		used_dose()
 
-/obj/item/organ/internal/heart/cybernetic/proc/used_dose()
-	owner.reagents.add_reagent(rid, ramount)
-	dose_available = FALSE
+	if(organ_flags & ORGAN_EMP)
+		return
+
+	if(stabilization_available && owner.health <= owner.crit_threshold)
+		stabilizer_heart()
+
+	if(bleed_prevention && ishuman(owner) && owner.blood_volume < BLOOD_VOLUME_NORMAL)
+		var/mob/living/carbon/human/wounded_owner = owner
+		wounded_owner.blood_volume += 2 * seconds_per_tick
+		if(toxification_probability && prob(toxification_probability))
+			wounded_owner.adjustToxLoss(1 * seconds_per_tick, updating_health = FALSE)
+
+		var/datum/wound/bloodiest_wound
+
+		for(var/datum/wound/iter_wound as anything in wounded_owner.all_wounds)
+			if(iter_wound.blood_flow && iter_wound.blood_flow > bloodiest_wound?.blood_flow)
+				bloodiest_wound = iter_wound
+
+		if(bloodiest_wound)
+			bloodiest_wound.adjust_blood_flow(-1 * seconds_per_tick)
+
+/obj/item/organ/internal/heart/cybernetic/proc/stabilizer_heart()
+	REMOVE_TRAIT(owner, TRAIT_NOSOFTCRIT, ORGAN_TRAIT)
+	stabilization_available = FALSE
+
+	addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_NOSOFTCRIT, ORGAN_TRAIT), stabilization_duration)
+
+	addtimer(VARSET_CALLBACK(src, stabilization_available, TRUE), 5 MINUTES)
```

Comment:
Does this timer clean up if the heart is deleted? I can never remember if timers have TIMER_DELETE_ME enabled by default.

Reply:
I'll check

Reply:
Well, just to be on the safe side I'll add it.